src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,252 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
+    this->actionFunc = BgMoriElevator_SetPosition;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1C40.s")
+void BgMoriElevator_SetPosition(BgMoriElevator* this, GlobalContext* globalCtx) {
+    s32 pad;
+    if (BgMoriElevator_IsLinkRiding(this, globalCtx)) {
+        if (globalCtx->roomCtx.curRoom.num == 2) {
+            this->targetYPos = -779.0f;
+            BgMoriElevator_StopMovement(this);
+        } else if (globalCtx->roomCtx.curRoom.num == 17) {
+            this->targetYPos = 233.0f;
+            BgMoriElevator_StopMovement(this);
+        } else {
+            // Error:Forest Temple obj elevator Room setting is dangerous(% s % d)
+            osSyncPrintf("Error : 森の神殿 obj elevator 部屋設定が危険(%s %d)\n", "../z_bg_mori_elevator.c", 479);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1CF4.s")
+    } else if ((globalCtx->roomCtx.curRoom.num == 2) && (this->dyna.actor.posRot.pos.y < -275.0f)) {
+        this->targetYPos = 233.0f;
+        BgMoriElevator_StopMovement(this);
+    } else if ((globalCtx->roomCtx.curRoom.num == 17) && (-275.0f < this->dyna.actor.posRot.pos.y)) {
+        this->targetYPos = -779.0f;
+        BgMoriElevator_StopMovement(this);
+    } else if (((globalCtx->roomCtx.curRoom.num == 2) &&
+                (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F))) &&
+               (this->unk_16C == 0)) {
+        this->targetYPos = 73.0f;
+        func_808A1C30(this);
+    } else if (((globalCtx->roomCtx.curRoom.num == 2)) &&
+               !(Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) && (this->unk_16C != 0)) {
+        this->targetYPos = 233.0f;
+        func_808A1CF4(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1D50.s")
+void BgMoriElevator_StopMovement(BgMoriElevator* this) {
+    this->actionFunc = func_808A2008;
+    this->dyna.actor.velocity.y = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1E04.s")
+void func_808A2008(BgMoriElevator* this, GlobalContext* globalCtx) {
+    f32 distTo;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1E14.s")
+    func_808A1800(&this->dyna.actor.velocity.y, 12.0f, 0.1f, 1.0f, 0.0f);
+    distTo = func_808A1800(&this->dyna.actor.posRot.pos.y, this->targetYPos, 0.1f, this->dyna.actor.velocity.y, 0.3f);
+    if (fabsf(distTo) < 0.001f) {
+        func_808A1E04(this);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_ELEVATOR_STOP);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1FF0.s")
+    } else {
+        func_808A18FC(this, distTo);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A2008.s")
+void BgMoriElevator_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriElevator* this = THIS;
```

Comment:
Minor, but we prefer having a space between var declaration and the rest of the block